test/test_utils.py

```diff
@@ -642,5 +645,56 @@ def test_assert_true(self):
             torch.Assert(False, "bar")
 
 
+class TestStandaloneCPPJIT(TestCase):
+    def test_load_standalone(self):
+        build_dir = tempfile.mkdtemp()
+        try:
+            src_path = os.path.join(build_dir, "main.cpp")
+            src = textwrap.dedent("""\
+                #include <iostream>
+                #include <torch/torch.h>
+                int main() {
+                    auto x = torch::eye(3);
+                    std::cout << x << std::endl;
+                }
+            """)
+            with open(src_path, "wt") as f:
+                f.write(src)
+
+            exec_path = torch.utils.cpp_extension.load(
+                "standalone_load_test",
+                src_path,
+                build_directory=build_dir,
+                is_python_module=False,
+                is_standalone=True,
+            )
+
+            ext = "exe" if IS_WINDOWS else "o"
+            self.assertEqual(
+                exec_path,
+                os.path.join(build_dir, f"standalone_load_test.{ext}")
+            )
+
+            for shell in [True, False]:
+                r = subprocess.run(
+                    [exec_path],
+                    shell=shell,
+                    stdout=subprocess.PIPE,
+                )
+                self.assertEqual(r.returncode, 0)
+                self.assertEqual(
```

Comment:
expect test?

Reply:
In this case no, because:
1) `expecttest` doesn't handle the loop properly. (Tries to double update.)
2) The replacement is completely dedented.

I tried it (factoring the run bits so there were two literal tests) and it was just ugly and awkward. If there were more test cases or the output was less trivial, yeah I'd convert it to an expect test.